src/daemon/devnode.c

```diff
@@ -36,6 +36,33 @@ int rm_recursive(const char* path){
     return remove(path);
 }
 
+int check_chown(const char *pathname, uid_t owner, gid_t group) {
+    if (chown(pathname, owner, group)) {
+        return 0;
+    } else {
+        ckb_warn("Chown call failed %s: %s\n", pathname, strerror(errno));
+        return -1;
+    }
+}
+
+int check_fchown(int fd, uid_t owner, gid_t group) {
+    if (fchown(fd, owner, group)) {
+        return 0;
+    } else {
+        ckb_warn("FChown call failed: %s\n", strerror(errno));
+        return -1;
+    }
+}
+
+int check_chmod(const char *pathname, mode_t mode) {
+    if(chmod(pathname, mode)) {
```

Comment:
See comments for `chown`. I'm not sure a warning suffices in this particular case, though; I'm not familiar with this particular code, but I'd lean towards making this fatal for possible security implications.